plugins/destination/plugin.go

```diff
@@ -3,54 +3,127 @@ package destination
 import (
 	"context"
 	"fmt"
+	"sync"
 	"time"
 
 	"github.com/cloudquery/plugin-sdk/schema"
 	"github.com/cloudquery/plugin-sdk/specs"
 	"github.com/rs/zerolog"
-	"golang.org/x/sync/errgroup"
+)
+
+type writerType int
+
+const (
+	unmanaged writerType = iota
+	managed
 )
 
 type NewClientFunc func(context.Context, zerolog.Logger, specs.Destination) (Client, error)
 
+type ManagedWriter interface {
+	PreWriteTableBatch(ctx context.Context, tables schema.Tables) error
```

Comment:
```suggestion
	InitializeTable(ctx context.Context, tables schema.Tables) error
```

I think `PreWriteTableBatch` might give the impression that this function will be called before every call to `WriteTableBatch`. Maybe we should call it something like `OpenTable` (and `CloseTable` for a post-hook). For BigQuery this might instantiate the client, for files this might open the file.

Reply:
I think yes, I'll do something like `InitTable` and `CloseTable`. Given that we don't use it now anywhere so I'll just drop this.